library/kotlin/src/io/envoyproxy/envoymobile/RetryPolicy.kt

```diff
@@ -25,18 +25,49 @@ data class RetryPolicy(
       throw IllegalArgumentException("Per-retry timeout cannot be less than total timeout")
     }
   }
+
+  /**
+   * Initialize the retry policy from a set of headers.
+   *
+   * @param headers: The headers with which to initialize the retry policy.
+   */
+  constructor(headers: Headers): this(
+      headers.value("x-envoy-max-retries")?.first()!!.toInt(),
```

Comment:
I think we should avoid crashing here like we do in Swift (and early return)

Reply:
Same as above, there is no concept of a failable constructor. So our option is to have a default value, or crash.

Reply:
https://github.com/lyft/envoy-mobile/pull/840#discussion_r425429624